lib/galaxy/util/__init__.py

```diff
@@ -1008,7 +1019,11 @@ def unicodify(value, encoding=DEFAULT_ENCODING, error='replace'):
     >>> s = u'lâtín strìñg'; assert unicodify(s.encode('latin-1')) == u'l\ufffdt\ufffdn str\ufffd\ufffdg'
     >>> s = u'lâtín strìñg'; assert unicodify(s.encode('latin-1'), error='ignore') == u'ltn strg'
     """
-    if value is None or isinstance(value, text_type):
+    if value is None:
+        return value
+    if isinstance(value, text_type):
+        if strip_null:
+            return value.replace('\0', '')
         return value
```

Comment:
If we want to simplify the code, we can remove the shortcut for `text_type` (i.e. lines 1024-1026).

Reply:
Good point, let's do that, I doubt this is performance critical.